### PR TITLE
Provide System.IO.Hashing package readme

### PR DESCRIPTION
**Pull request by eNeRGy164**
Fill out details for the `System.IO.Hashing` NuGet package readme.

See #92228

### CODE REVIEW

File: src/libraries/System.IO.Hashing/src/PACKAGE.md [hunk 1]
```diff
@@ -2,43 +2,87 @@
 
 <!-- A description of the package and where one can find more documentation -->
 
+System.IO.Hashing offers a variety of hash code algorithms.
 
+Hash code algorithms are pivotal for generating unique values for objects based on their content, facilitating object comparisons, and detecting content alterations.
```
Comment (saul) [inline src/libraries/System.IO.Hashing/src/PACKAGE.md:7]: Is it worth putting "these hash functions should not be used for security - e.g. passwords or verifying untrusted content" and linking to the System.Security.Cryptography much higher up?

File: src/libraries/System.IO.Hashing/src/PACKAGE.md [hunk 2]
```diff
@@ -2,43 +2,87 @@
...
-* [Conceptual documentation](https://learn.microsoft.com/en-us/dotnet/standard/serialization/**LIBRARYNAME**/overview)
-* [API documentation](https://learn.microsoft.com/en-us/dotnet/api/**LIBRARYNAME**)
+* [xxHash - Extremely fast hash algorithm](https://github.com/Cyan4973/xxHash/blob/release/doc/xxhash_spec.md)
+* [API documentation](https://learn.microsoft.com/en-us/dotnet/api/system.io.hashing)
 
 ## Related Packages
 
 <!-- The related packages associated with this package -->
 
+Cryptographic services, including secure encoding and decoding of data: [System.Security.Cryptography](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography)
```
Comment (saul) [inline src/libraries/System.IO.Hashing/src/PACKAGE.md:82]:
```suggestion
Cryptographic services, including secure encryption and decryption of data: [System.Security.Cryptography](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography)
```

File: src/libraries/System.IO.Hashing/src/PACKAGE.md (hunk 1 above)
Comment (eNeRGy164) [reply src/libraries/System.IO.Hashing/src/PACKAGE.md:7]: Great suggestion, will add that

**Comment by ViktorHofer** [comment]
@eNeRGy164 looks like your branch has a conflict (it adds 523 commits)

**Comment by eNeRGy164** [comment]
> @eNeRGy164 looks like your branch has a conflict (it adds 523 commits)

Yes, I pressed the for sync button on the wrong branch 😭 
Will fix this in a second

File: src/libraries/System.IO.Hashing/src/PACKAGE.md [hunk 3]
```diff
@@ -2,43 +2,90 @@
...
 
 ## Key Features
 
 <!-- The key features of this package -->
 
-*
-*
-*
+* Variety of hash code algorithms including CRC-32, CRC-64, xxHash3, xxHash32, xxHash64, and xxHash128.
+* Implementations of CRC-32 and CRC-64 algorithms, as used in ITU-T V.42, IEEE 802.3, and described in ECMA-182, Annex B respectively.
```
Comment (bartonjs) [inline src/libraries/System.IO.Hashing/src/PACKAGE.md:18]: A two-item list is compared to a three-item list respectively.

Perhaps two and two?

```suggestion
* Implementations of CRC-32 and CRC-64 algorithms, as used in IEEE 802.3, and described in ECMA-182, Annex B respectively.
```

File: src/libraries/System.IO.Hashing/src/PACKAGE.md [hunk 4]
```diff
@@ -2,43 +2,90 @@
...
+* `XxHash128`
 
 ## Additional Documentation
 
 <!-- Links to further documentation. Remove conceptual documentation if not available for the library. -->
 
-* [Conceptual documentation](https://learn.microsoft.com/en-us/dotnet/standard/serialization/**LIBRARYNAME**/overview)
-* [API documentation](https://learn.microsoft.com/en-us/dotnet/api/**LIBRARYNAME**)
+* [xxHash - Extremely fast hash algorithm](https://github.com/Cyan4973/xxHash/blob/release/doc/xxhash_spec.md)
+* [API documentation](https://learn.microsoft.com/en-us/dotnet/api/system.io.hashing)
```
Comment (bartonjs) [inline src/libraries/System.IO.Hashing/src/PACKAGE.md:79]: I'd flip these, API docs before docs talking about one of the hash families in the package.

**Review by bartonjs — APPROVED** [review]

**Comment by bartonjs** [comment]
@eNeRGy164 By the by, it's generally frowned upon to do a squash/amend during a PR (anything requiring a force-push).  It makes it harder to review what got changed as a response to the feedback.  (We squash it up when merging)

Yeah, there's a github feature to look at what changed in a force push, but doesn't exist in the `Files changed`.  But, in general, the preferred form is to leave the individual commits once a PR has started.

**Comment by eNeRGy164** [comment]
> @eNeRGy164 By the by, it's generally frowned upon to do a squash/amend during a PR (anything requiring a force-push). It makes it harder to review what got changed as a response to the feedback. (We squash it up when merging)

Thanks for pointing that out. I didn't notice the mention on squash commit.
I'm used to work at repos where clean git history is valued, and one should not create review commits.

**Pull request merged**